lib/NormalModule.js

```diff
@@ -125,19 +127,32 @@ const asBuffer = input => {
 	return input;
 };
 
-const readResourceFn = fs => {
-	return (resource, callback) => {
-		const decodedData = decodeDataURI(resource);
-
-		if (decodedData) {
-			process.nextTick(() => {
-				callback(null, decodedData);
-			});
-			return;
+/**
+ * @param {InputFileSystem} fs fs
+ * @param {string|undefined} scheme scheme
+ * @returns {(resource: string, cb: (err: Error|null, data?: string|Buffer) => void) => void} read function
+ */
+const readResourceFn = (fs, scheme) => {
```

Comment:
loaders could potentially change the `resource` in pitch mode. So you better parse the schema again and don't take it from before.